### PR TITLE
expose welcomeScreen on Invite

### PR DESCRIPTION
**Pull request by Khaaz**
discord/discord-api-docs#1636

### CODE REVIEW

File: lib/structures/Invite.js [hunk 1]
```diff
@@ -45,7 +48,18 @@ class Invite extends Base {
                 icon: data.guild.icon,
                 features: data.guild.features,
                 verificationLevel: data.guild.verification_level,
-                vanityUrlCode: data.guild.vanity_url_code
+                vanityUrlCode: data.guild.vanity_url_code,
+                welcomeScreen: data.guild.welcome_screen !== undefined && {
```
Comment (apacheli) [inline lib/structures/Invite.js]: `welcomeScreen` can end up as a boolean here when it shouldn't.

**Review by apacheli — CHANGES_REQUESTED** [review]

File: index.d.ts [hunk 2]
```diff
@@ -1725,6 +1725,16 @@ declare namespace Eris {
...
+  interface WelcomeChannel {
+	channelID: string;
+	description: string;
+	emojiID: string;
+	emojiName: string;
+  }
+  interface WelcomeScreen {
+	description: string;
+	welcomeChannels: WelcomeChannel[];
+  }
```
Comment (bsian03) [inline index.d.ts]: Needs to go in its relevant interface group a little while up

File: index.d.ts [hunk 3]
```diff
@@ -1776,7 +1786,8 @@ declare namespace Eris {
     verificationLevel: number;
     voiceStates: Collection<VoiceState>;
     widgetChannelID?: string | null;
-    widgetEnabled?: boolean | null;
+	widgetEnabled?: boolean | null;
+	welcomeScreen?: WelcomeScreen;
```
Comment (bsian03) [inline index.d.ts]: indents
```suggestion
    widgetEnabled?: boolean | null;
    welcomeScreen?: WelcomeScreen;
```

File: lib/structures/Guild.js [hunk 4]
```diff
@@ -66,6 +66,9 @@ const {Permissions} = require("../Constants");
 * @prop {Collection<VoiceState>} voiceStates Collection of voice states in the guild
 * @prop {Number?} widgetChannelID The channel id that the widget will generate an invite to. REST only.
 * @prop {Boolean?} widgetEnabled Whether the guild widget is enabled. REST only.
+* @prop {Object?} guild.welcomeScreen The welcome screen of a Community guild, shown to new members
+* @prop {Object} guild.welcomeScreen.description The description in the welcome screen
+* @prop {Array<Object>} guild.welcomeScreen.welcomeChannels The list of channels in the welcome screens. Each channels have the following properties: `channelID`, `description`, `emojiID`, `emojiName`. Theses properties can be null.
```
Comment (bsian03) [inline lib/structures/Guild.js]:
```suggestion
* @prop {Array<Object>} guild.welcomeScreen.welcomeChannels The list of channels in the welcome screens. Each channels have the following properties: `channelID`, `description`, `emojiID`, `emojiName`. `emojiID` and `emojiName` properties can be null.
```

Comment (bsian03) [inline lib/structures/Guild.js]: Alphabetical sorting

File: index.d.ts [hunk 5]
```diff
@@ -1725,6 +1725,16 @@ declare namespace Eris {
     category_id: number;
   }
 
+  interface WelcomeChannel {
+	channelID: string;
+	description: string;
+	emojiID: string;
+	emojiName: string;
```
Comment (bsian03) [inline index.d.ts]: See JS review 
```suggestion
	emojiID: string | null;
	emojiName: string | null;
```

**Review by bsian03 — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 2 above)
Comment (Khaaz) [reply index.d.ts]: That seems to belong here though. This is a Guild interface isn't it?

**Review by abalabahaha — DISMISSED** [review]

File: lib/structures/Guild.js [hunk 6]
```diff
@@ -64,9 +64,12 @@ const {Permissions} = require("../Constants");
 * @prop {String?} vanityURL The vanity URL of the guild (VIP only)
 * @prop {Number} verificationLevel The guild verification level
 * @prop {Collection<VoiceState>} voiceStates Collection of voice states in the guild
+* @prop {Object?} welcomeScreen The welcome screen of a Community guild, shown to new members
+* @prop {Object} welcomeScreen.description The description in the welcome screen
+* @prop {Array<Object>} welcomeScreen.welcomeChannels The list of channels in the welcome screens. Each channels have the following properties: `channelID`, `description`, `emojiID`, `emojiName`. `emojiID` and `emojiName` properties can be null.
 * @prop {Number?} widgetChannelID The channel id that the widget will generate an invite to. REST only.
 * @prop {Boolean?} widgetEnabled Whether the guild widget is enabled. REST only.
-*/
+**/
```
Comment (abalabahaha) [inline lib/structures/Guild.js]:
```suggestion
*/
```
ytho

**Review by abalabahaha — APPROVED** [review]

**Pull request merged**